src/components/biggive-campaign-card-filter-grid/biggive-campaign-card-filter-grid.tsx

```diff
@@ -179,9 +180,19 @@ export class BiggiveCampaignCardFilterGrid {
   }
 
   private handleApplyFilterButtonClick = () => {
-    this.doSearchAndFilterUpdate.emit(this.getSearchAndFilterObject());
-    console.log(this.el.shadowRoot.getElementById('filter-popup'));
+    const searchAndFilterObj = this.getSearchAndFilterObject();
+    this.doSearchAndFilterUpdate.emit(searchAndFilterObj);
     this.el.shadowRoot.getElementById('filter-popup').closeFromOutside();
+
+    const anyFilterApplied: boolean =
+      searchAndFilterObj.filterBeneficiary !== undefined ||
+      searchAndFilterObj.filterCategory !== undefined ||
+      searchAndFilterObj.filterFunding !== undefined ||
+      searchAndFilterObj.filterLocation !== undefined;
```

Comment:
my ts is a not that strong, but I'm a bit confused by the types here. Isn't e.g. filter location defined as being string type at https://github.com/thebiggive/components/blob/90b461aa458a05009f074c1c52bf09b40b6ab7c1/src/components/biggive-campaign-card-filter-grid/biggive-campaign-card-filter-grid.tsx#L135 ? If it's string how can it be undefined?

Reply:
ah I see, I forgot [that](https://www.typescriptlang.org/docs/handbook/advanced-types.html)
> By default, the type checker considers null and undefined assignable to anything. Effectively, null and undefined are valid values of every type. That means it’s not possible to stop them from being assigned to any type, even when you would like to prevent it. The inventor of null, Tony Hoare, calls this his [“billion dollar mistake”](https://wikipedia.org/wiki/Null_pointer#History).